vernac/comFixpoint.ml

```diff
@@ -224,10 +224,16 @@ let interp_recursive ~program_mode ~cofix fixl notations =
 
   (* Instantiate evars and check all are resolved *)
   let sigma = solve_unif_constraints_with_heuristics env_rec sigma in
-  let sigma = Evd.minimize_universes sigma in
-  (* XXX: We still have evars here in Program *)
-  let fixdefs = List.map (fun c -> Option.map EConstr.(to_constr ~abort_on_undefined_evars:false sigma) c) fixdefs in
-  let fixtypes = List.map EConstr.(to_constr sigma) fixtypes in
+  let sigma, (rec_sign, fixdefs, fixtypes) =
+    (* NB: we still have evars in program mode *)
+    Evarutil.finalize ~abort_on_undefined_evars:false
+      env sigma
+      (fun nf ->
+         let fixdefs = List.map (fun c -> Option.map nf c) fixdefs in
+         let fixtypes = List.map nf fixtypes in
+         let rec_sign = List.map (map_constr (fun c -> EConstr.of_constr (nf c))) rec_sign in
+         rec_sign, fixdefs, fixtypes)
+  in
```

Comment:
Here we restrict even though we didn't use to in program mode (it was done at a different location in normal mode).
Also we nf rec_sign which didn't use to be done.